systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java

```diff
@@ -1324,9 +1328,68 @@ void testConfigureDeploymentStrategy(ExtensionContext extensionContext) {
         assertThat(kafkaConnect.getSpec().getTemplate().getDeployment().getDeploymentStrategy(), is(DeploymentStrategy.ROLLING_UPDATE));
     }
 
+    @ParallelNamespaceTest
+    void testKafkaAndKafkaConnectWithJMX(ExtensionContext extensionContext) {
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String kafkaClientsName = mapWithKafkaClientNames.get(extensionContext.getDisplayName());
+        final String namespace = extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(Constants.NAMESPACE_KEY).toString();
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3)
+            .editOrNewSpec()
+                .editKafka()
+                    .withNewJmxOptions()
+                        .withAuthentication(new KafkaJmxAuthenticationPassword())
+                    .endJmxOptions()
+                .endKafka()
+            .endSpec()
+            .build());
+
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(false, kafkaClientsName).build());
+        String clientsPodName = kubeClient().listPodsByPrefixInName(kafkaClientsName).get(0).getMetadata().getName();
+
+        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(extensionContext, clusterName, 1, true)
+            .editOrNewSpec()
+                .withNewJmxOptions()
+                    .withAuthentication(new KafkaJmxAuthenticationPassword())
+                .endJmxOptions()
+            .endSpec()
+            .build());
+
+        Map<String, Map<String, String>> resources = new HashMap<>();
+
+        Map<String, String> kafkaMap = new HashMap<>();
+        kafkaMap.put("service-name", KafkaResources.brokersServiceName(clusterName));
+        kafkaMap.put("secret-name", clusterName + "-kafka-jmx");
+        kafkaMap.put("cmd", "bean kafka.server:type=app-info\nget -i *");
+
+        Map<String, String> kafkaConnectMap = new HashMap<>();
+        kafkaConnectMap.put("service-name", KafkaConnectResources.serviceName(clusterName));
+        kafkaConnectMap.put("secret-name", clusterName + "-kafka-connect-jmx");
+        kafkaConnectMap.put("cmd", "bean kafka.connect:type=app-info\nget -i *");
+
+        resources.put(Kafka.RESOURCE_KIND, kafkaMap);
+        resources.put(KafkaConnect.RESOURCE_KIND, kafkaConnectMap);
+
+        resources.forEach((resource, resourceMap) -> {
+            String serviceName = resourceMap.get("service-name");
+            String secretName = resourceMap.get("secret-name");
+            Secret jmxSecret = kubeClient(namespace).getSecret(secretName);
```

Comment:
This looks real ugly. Maybe having some helper class to hold these configuration would make it nicer? It can be private, just within the test class itself or maybe it can live with the JMXUtils and be used there as well.